ch5/lkm_template/Makefile

```diff
@@ -35,9 +35,12 @@
 # This enables you to use this Makefile as a template; just update this variable!
 # As well, the MYDEBUG variable (see it below) can be set to 'y' or 'n' (no being
 # the default)
-FNAME_C := lkm_template
+FNAME_C ?= lkm_template
 ifeq ($(FNAME_C),)
-  $(error ERROR: you Must set the FNAME_C variable in the Makefile)
+  $(error ERROR: you Must pass the C file like this: \
+  make FNAME_C=csrc-filename-without-.c target-name)
+else ifneq ("$(origin FNAME_C)", "command line")
```

Comment:
very cool! i was unaware of this.. thx

Reply:
Glad I could be of any help. Thank you sir.